common/service/config/archival.go

```diff
@@ -50,6 +49,6 @@ func isArchivalConfigValid(
 	URISet := len(domianDefaultURI) != 0
 
 	validEnable := archivalEnabled && URISet && specifiedProvider
-	validDisabled := !archivalEnabled && !enableRead && domainDefaultStatus != common.ArchivalEnabled && !URISet && !specifiedProvider
+	validDisabled := !archivalEnabled && !enableRead && domainDefaultStatus != common.ArchivalEnabled && !URISet
```

Comment:
I have to remove this check, otherwise there is no way to workaround in the config_template. 
And I don't think it's necessary to enforce this. LMK if I am wrong.

Reply:
@andrewjdawson2016 LMK if you have concern about that. We could emit a warn log if you think this is useful to prevent users making mistake in configuration.

Reply:
I believe it should be fine :-)  So I am going to merge it in.